pennylane/circuit_drawer/representation_resolver.py

```diff
@@ -119,7 +119,7 @@ def single_parameter_representation(self, par):
         if isinstance(par, str):
             return par
 
-        return str(round(par, 3))
+        return f"{1.0 * par:.3g}"
```

Comment:
This is required since tape parameters can now be TF tensors or Torch tensors in addition to floats/NumPy arrays. Unfortunately `round()` doesn't work with Torch tensors, so we instead use string formatting.

The multiplication by `1.0` is to convert a TF Variable into a TF tensor without importing tensorflow. This is because TensorFlow Variables do not support string formatting for some reason 😢